packages/framework/azure-client/src/AzureClient.ts

```diff
@@ -73,29 +82,61 @@ export class AzureClient {
             config: {},
         });
 
-        const rootDataObject = await requestFluidObject<RootDataObject>(
+        const fluidContainer = await this.createFluidContainer(
             container,
-            "/",
+            this.props.connection,
         );
-        const createNewRequest = createAzureCreateNewRequest(
-            this.props.connection.orderer,
-            this.props.connection.storage,
-            this.props.connection.tenantId,
+        const services = this.getContainerServices(container);
+        return { container: fluidContainer, services };
+    }
+
+    /**
+     * Recreates new container out of specific version of another container
+     * @param id - Unique ID of the source container in Azure Fluid Relay.
+     * @param version - Unique version of the source container in Azure Fluid Relay.
+     * @param containerSchema - Container schema used to access data objects in the container.
+     * @returns New container instance along with associated services.
+     */
+    public async reCreateContainer(
+        id: string,
+        version: string,
```

Comment:
I think this will work better as an optional object property on the createContainer call. I would love to go with a design that is more extensible to other possible creation flows down the road.

Did you consider this already with your design?

Here's an example of what I'm thinking of using type guards to differentiate between other potential future types.

The down sides I can see:
- It's more cognitive load for the developer going down this path but
  - I think this is fine considering this is a specialized scenario.
- If we want to extend create with additional options (for whatever reason) this could paint us in a corner.
  - I think another object layer would be the answer here but it's crazy overkill considering I don't have anything in mind that we would need here.

```typescript
interface ReCreateContainerOption {
  type: "recreate",
  id: string,
  version: string,
}

function isReCreateContainerOption(containerOption: any): containerOption is ReCreateContainerOption {
    return containerOption.type === "recreate";
}

// can add more types in the future
type AzureCreateContainerOption = ReCreateContainerOption; 

public async createContainer(containerSchema: ContainerSchema, createOption?:AzureCreateContainerOption) {
 if (isReCreateContainerOption(containerOption)) {
    // ...
 }
}
```

Reply:
note. I didn't apply my naming proposal here to not confuse the two conversations.

Reply:
As we discussed offline. Let's go with the new `copyContainer` method approach. If it turns out that there are a collection of these types of functions we can concatenate as part of an API refactor.

Reply:
I have renamed the function to `copyContainer`. It is still exposed directly through Azure Client and not through `AzureContainerServices`. Agree, let's see where this API extends, use cases that drive them, and then we can reason about complexity vs value.